app/src/main/java/org/simple/clinic/recentpatientsview/RecentPatientsView.kt

```diff
@@ -55,23 +52,26 @@ class RecentPatientsView(context: Context, attrs: AttributeSet) : FrameLayout(co
 
     recyclerView.apply {
       layoutManager = LinearLayoutManager(context)
-      adapter = groupAdapter
+      adapter = recentAdapter
       isNestedScrollingEnabled = false
     }
 
     bindUiToController(
         ui = this,
-        events = Observable.merge(screenCreates(), adapterEvents),
+        events = Observable.merge(screenCreates(), adapterEvents()),
         controller = controller,
-        screenDestroys = RxView.detaches(this).map { ScreenDestroyed() }
+        screenDestroys = detaches.map { ScreenDestroyed() }
     )
   }
 
   private fun screenCreates() = Observable.just(ScreenCreated())
 
-  fun updateRecentPatients(recentPatients: List<RecentPatientItemType<out ViewHolder>>) {
-    recentPatients.forEach { it.uiEvents = adapterEvents }
-    groupAdapter.update(recentPatients)
+  private fun adapterEvents() = recentAdapter.itemEvents
+      .ofType<UiEvent>()
+      .takeUntil(detaches)
```

Comment:
This is not necessary since `adapterEvents` is already being passed to the controller, which will dispose it when the screen is detached.

Reply:
fixed it. So if I got it right, all the observable events that are passed to `MobiusDelegate` and controller will automatically get disposed when screen is detached?

Reply:
Yes.